### PR TITLE
Financial Connections: for networking manual entry, moved AccountUpdateRequired drawer to use GenericInfoVC

### PR DESCRIPTION
**Pull request by kgaidis-stripe**
## Summary

This PR moves and deletes `AccountUpdateRequiredViewController` in favor of backend and `GenericInfoViewController`.

Based off [Stripe.js PR](https://git.corp.stripe.com/stripe-internal/stripe-js-v3/pull/23288)

## Testing

Previously `bitrise run financial-connections-stability-tests` was failing, now it's passing:

```
FinancialConnectionsNetworkingUITests
    ✓ testNativeNetworkingTestMode (143.045 seconds)
    ✓ testNativeNetworkingTestModeSignUpWithMultiSelectAndPrefilledEmail (31.740 seconds)
    ✓ testNativeNetworkingTestModeSignUpWithPrefilledEmailAndPhone (24.437 seconds)
FinancialConnectionsUITests
    ✓ testDataLiveModeOAuthNativeAuthFlow (28.850 seconds)
    ✓ testDataLiveModeOAuthWebAuthFlow (19.323 seconds)
    ✓ testDataTestModeOAuthNativeAuthFlow (20.381 seconds)
    ✓ testPaymentSearchInLiveModeNativeAuthFlow (25.408 seconds)
    ✓ testPaymentTestModeLegacyNativeAuthFlow (19.474 seconds)
    ✓ testPaymentTestModeManualEntryAutofill (15.112 seconds)
    ✓ testPaymentTestModeManualEntryNativeAuthFlow (24.090 seconds)
    ✓ testWebInstantDebitsFlow (12.847 seconds)

```

Video of the drawer:

https://github.com/user-attachments/assets/c7b6aead-737d-47a7-a06f-5121384debaf

### CODE REVIEW

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift [hunk 1]
```diff
@@ -424,108 +433,124 @@ extension LinkAccountPickerViewController: LinkAccountPickerBodyViewDelegate {
...
-                            .analyticsClient
-                            .logUnexpectedError(
-                                FinancialConnectionsSheetError
-                                    .unknown(
-                                        debugDescription: "Updating a repair account, but repairs are not supported in Mobile."
-                                    ),
-                                errorName: "UpdateRepairAccountError",
-                                pane: .linkAccountPicker
+
+                    let genericInfoViewController = GenericInfoViewController(
```
Comment (kgaidis-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift:508]: This is a lot easier to review if whitespaces are removed:

Screenshot 2024-07-25 at 2 15 37 PM

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift [hunk 2]
```diff
@@ -424,108 +433,124 @@ extension LinkAccountPickerViewController: LinkAccountPickerBodyViewDelegate {
...
+                        genericInfoScreen: drawerOnSelection,
+                        theme: dataSource.manifest.theme,
+                        panePresentationStyle: .sheet,
+                        iconView: {
+                            let institutionIconView = InstitutionIconView()
+                            institutionIconView.setImageUrl(selectedPartnerAccount.institution?.icon?.default)
+                            return institutionIconView
+                        }(),
```
Comment (kgaidis-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift:516]: we override the backend icon because the icon is dependent on selection (and backend doesn't know the selection):

[Stripe.js override](https://git.corp.stripe.com/stripe-internal/stripe-js-v3/pull/23288/files#diff-b2beff986072eb286873f4aea4307fedbe4b05a9c4b1e5328fb1cded1253b1f3R35)

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift [hunk 3]
```diff
@@ -424,108 +433,124 @@ extension LinkAccountPickerViewController: LinkAccountPickerBodyViewDelegate {
...
+                        },
+                        // "did select cancel"
+                        didSelectSecondaryButton: { genericInfoViewController in
+                            delayDeselectingAccounts = false
+                            genericInfoViewController.dismiss(
+                                animated: true
+                            )
+                        },
+                        didSelectURL: { _ in
+                            // TODO(kgaidis): fix handling URL
```
Comment (kgaidis-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift:538]: this will be improved in another PR (same note applies in another part)

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift [hunk 4]
```diff
@@ -354,18 +353,28 @@ extension LinkAccountPickerViewController: LinkAccountPickerBodyViewDelegate {
...
-                genericInfoScreen: drawerOnSelection,
-                theme: dataSource.manifest.theme,
-                panePresentationStyle: .sheet,
-                didSelectPrimaryButton: { genericInfoViewController in
-                    genericInfoViewController.dismiss(animated: true)
-                },
-                didSelectURL: { _ in } // TODO(kgaidis): fix handling URL
-            )
-            genericInfoViewController.present(on: self)
+            if !eligibleToPresentAccountUpdateRequiredDrawer {
```
Comment (kgaidis-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift:364]: [Stripe.js](https://git.corp.stripe.com/stripe-internal/stripe-js-v3/pull/23288/files#diff-96f25fd92e7a6ede67b5de1870616635186357cb3642022af8a41e0c25a968faR217)

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift [hunk 5]
```diff
@@ -354,18 +353,28 @@ extension LinkAccountPickerViewController: LinkAccountPickerBodyViewDelegate {
         FeedbackGeneratorAdapter.selectionChanged()
 
         let selectedPartnerAccount = selectedAccountTuple.partnerAccount
+        let eligibleToPresentAccountUpdateRequiredDrawer: Bool = (
+            // repair flow
+            selectedPartnerAccount.nextPaneOnSelection == .bankAuthRepair
+                // supportability -- account requires re-sharing with additonal permissions
+                || selectedPartnerAccount.nextPaneOnSelection == .partnerAuth
+        )
```
Comment (kgaidis-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift:361]: this logic was moved from the bottom of the file, but [stripe.js here also](https://git.corp.stripe.com/stripe-internal/stripe-js-v3/pull/23288/files#diff-96f25fd92e7a6ede67b5de1870616635186357cb3642022af8a41e0c25a968faR217) and [here too](https://git.corp.stripe.com/stripe-internal/stripe-js-v3/blob/b29037a9/src/linkedAccounts/inner/components/panes/v3/LinkAccountPickerPane/lib/helpers.ts#L3-L12)

File: Example/FinancialConnections Example/FinancialConnectionsUITests/FinancialConnectionsNetworkingUITests.swift [hunk 6]
```diff
@@ -260,8 +260,8 @@ final class FinancialConnectionsNetworkingUITests: XCTestCase {
 
         app.fc_nativeBankAccount(name: bankAccountName).tap()
 
-        let accountUpdateRequiredContinueButton = app.buttons["account_update_required_continue_button"]
-        XCTAssertTrue(accountUpdateRequiredContinueButton.waitForExistence(timeout: 1))
+        let accountUpdateRequiredContinueButton = app.buttons["generic_info_primary_button"]
```
Comment (kgaidis-stripe) [inline Example/FinancialConnections Example/FinancialConnectionsUITests/FinancialConnectionsNetworkingUITests.swift:263]: this test was failing because `account_update_required_continue_button` was not being hit...the "generic info drawer" was being hit due to previous logic

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift (hunk 2 above)
Comment (mats-stripe) [reply StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift:516]: What happens here if `selectedPartnerAccount.institution?.icon?.default` is nil? Should we maybe only return the `institutionIconView` here if that institution icon isn't nil?

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift [hunk 7]
```diff
@@ -424,108 +433,124 @@ extension LinkAccountPickerViewController: LinkAccountPickerBodyViewDelegate {
...
-                        )
-                } else {
-                    dataSource
-                        .analyticsClient
-                        .log(
-                            eventName: "click.supportability_account",
-                            pane: .linkAccountPicker
-                        )
-                }
+            if  eligibleToPresentAccountUpdateRequiredDrawer {
```
Comment (mats-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift:436]: 👀 
```suggestion
            if eligibleToPresentAccountUpdateRequiredDrawer {
```

**Review by mats-stripe — APPROVED** [review]
LGTM!

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift (hunk 2 above)
Comment (kgaidis-stripe) [reply StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift]: ty improved here:

https://github.com/stripe/stripe-ios/pull/3833/commits/dc24a89977bceada6a613d95ce34e3c14f1b3560

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift (hunk 7 above)
Comment (kgaidis-stripe) [reply StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift]: ty fixed here: https://github.com/stripe/stripe-ios/pull/3833/commits/a6465b6b1230ee004fb88ef334b46c22716ce7f8
